commons/zenoh-buffers/src/zslice.rs

```diff
@@ -30,17 +31,59 @@ use core::{
 /*************************************/
 pub trait ZSliceBuffer: Send + Sync + fmt::Debug {
     fn as_slice(&self) -> &[u8];
+
+    #[cfg(feature = "shared-memory")]
+    /// # Safety
+    /// Get mutable access to underlying data. This is safe if
+    /// there is no sharing of underlying data.
+    unsafe fn as_mut_slice_unchecked(&mut self) -> &mut [u8];
+    #[cfg(feature = "shared-memory")]
+    fn as_mut_slice(&mut self) -> Option<&mut [u8]>;
+
+    #[cfg(not(feature = "shared-memory"))]
     fn as_mut_slice(&mut self) -> &mut [u8];
+
     fn as_any(&self) -> &dyn Any;
+    #[cfg(feature = "shared-memory")]
+    fn is_valid(&self) -> bool {
```

Comment:
A slice is supposed to be always valid, what's the point of this function? And what the user should do in case the slice is not valid?

Reply:
On SHM it appeals to Generation Counter mechanism, so ZSlice's contents are not always valid for `#[cfg(feature = "shared-memory")]`